bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java

```diff
@@ -266,9 +266,17 @@ public void processResult(int rc, String path, Object ctx, String name) {
                 }
             }
         };
+        final byte[] data;
+        try {
+            data = serDe.serialize(metadata);
+        } catch (IOException ioe) {
+            promise.completeExceptionally(new BKException.ZKException(ioe));
```

Comment:
I would suggest defining a new error code for this type of exception. `BKException.ZKException` is a bit confusing.

Reply:
Will add